addon/components/input-mask.js

```diff
@@ -0,0 +1,26 @@
+/* global moment */
+
+import Ember from 'ember';
+
+export default Ember.TextField.extend({
+  initializeMask: function() {
+    var mask = this.get('mask');
+
+    this.$().inputmask(mask, {
+      onBeforeMask: function(value) {
+        if (mask === 'mm/dd/yyyy') {
+          return moment(new Date(value)).format('L');
+        }
+      }
+    });
+
+    // The input mask changes the value of the input from the original to a
+    // formatted version. We need to manually send that change back to the
+    // controller.
+    this.set('value', this.$().val());
+  }.on('didInsertElement'),
+
+  removeMask: function() {
+    this.$().inputmask('remove');
```

Comment:
I'm not entirely positive. But if I don't have this I see the same number of event listeners before _and_ after the view is destroyed.

To test I created an app that uses this input in a view and toggle a conditional with the 'Remove' button that takes the input out of the DOM.

### with inputmask('remove')
![screen shot 2015-01-25 at 3 14 33 pm](https://cloud.githubusercontent.com/assets/8364/5893082/33d0b5d4-a4a5-11e4-9054-2b0706729f98.png)

### without inputmask('remove')
![screen shot 2015-01-25 at 3 12 39 pm](https://cloud.githubusercontent.com/assets/8364/5893084/3ac2f924-a4a5-11e4-88c6-abdf659f2645.png)

Without unbinding the inputmask plugin, the number of listeners increases with every insert-destroy cycle. When it is unbound, the listeners go up and down depending on whether or not the input is in the DOM.